### PR TITLE
PKG-14995: Update xarray to 2026.4.0

### PR DESCRIPTION
**Pull request by cbouss**
## xarray 2026.4.0

**Destination channel:** defaults

### Links
- [PKG-14995](https://anaconda.atlassian.net/browse/PKG-14995)
- [PyPI](https://pypi.org/project/xarray/2026.4.0/)
- [GitHub](https://github.com/pydata/xarray)

### Explanation of changes:
- Updated xarray from 2026.2.0 to 2026.4.0
- Removed `numpy <2.4` cap — xarray 2026.4.0 was released after numpy 2.4 and upstream does not cap it; the previous test failures (test_curvefit_helpers, test_concat_periods) should be resolved
- Bumped `packaging` from `>=24.1` to `>=24.2` (matching upstream)
- Update run_constrained

[PKG-14995]: https://anaconda.atlassian.net/browse/PKG-14995?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by cbouss** [comment]
```
pkg-review: ✓ 10 passed  ✗ 1 failed  ⚠ 2 warnings  (13 total)

⚠ artifact-size-regression: No prior artifact data available
⚠ cbc-dep-pinning: numpy >=1.26 hardcoded vs CBC (pre-existing pattern)
ℹ version-only-bump: Non-version changes: deps (packaging bump, numpy cap removed)

Clean — no actionable issues.
```

File: recipe/meta.yaml
Comment (cbouss) [inline recipe/meta.yaml]:
```suggestion
# accel
- scipy >=1.13
- numbagg >=0.9
- numba >=0.62
- flox >=0.10
# io
- netcdf4 >=1.6.0
- h5netcdf >=1.5.0
- zarr >=3.0
# etc
- sparse >=0.15
# viz
- cartopy >=0.23
- matplotlib-base >=3.8
```
https://github.com/pydata/xarray/blob/v2026.04.0/pyproject.toml#L28

**Comment by package-build-platform** [comment]
<details><summary>Failed to submit graph for 7f11ca584c510b7d6b5c1a67d899758bf3c436d9
</summary>Message:
Failed to run command: 'pbp generate from-feedstock --output /tmp/tmpa7ea8u5e/xarray-feedstock.yaml --custom-metadata-file /tmp/tmpa7ea8u5e/custom_metadata.yaml --release-name xarray-feedstock-17-7f11ca584c510b7d6b5c1a67d899758bf3c436d9 xarray-feedstock/recipe', with error: 'INFO:root:Found feedstock xarray-feedstock at /tmp/tmpa7ea8u5e/xarray-feedstock/recipe
INFO:root:Build/Test graph from xarray-feedstock
Created by sbx_user1051 on 2026-06-01 20:46:39
Command line: /opt/conda/envs/webhook-listener/bin/pbp generate from-feedstock --output /tmp/tmpa7ea8u5e/xarray-feedstock.yaml --custom-metadata-file /tmp/tmpa7ea8u5e/custom_metadata.yaml --release-name xarray-feedstock-17-7f11ca584c510b7d6b5c1a67d899758bf3c436d9 xarray-feedstock/recipe

INFO:root:Creating graph from feedstock at: xarray-feedstock
INFO:pbpcli.utils.common:Setting env var: CONDA_SUBDIR = linux-64.
INFO:pbpcli.utils.expand_variants:Elapsed time for get_package_variants: 0.11 seconds
INFO:pbpcli.utils.expand_variants:Matrix vars: 4 unchanging vars:(465), changing vars (4): ['blas_impl', 'cuda_compiler_version', 'numpy', 'python']
INFO:pbpcli.utils.expand_variants:Elapsed time for preparation for render: 0.00 seconds
INFO:pbpcli.utils.common:Setting env var: CONDA_PKGS_DIRS = /tmp/conda_pkgs_cjevqqyh.
Problematic recipe:



package:
  name: xarray
  version: 2026.4.0

source:
  url: https://pypi.org/packages/source/x/xarray/xarray-2026.4.0.tar.gz
  sha256: c4ac9a01a945d90d5b1628e2af045099a9d4943536d4f2ee3ae963c3b222d15b

build:
  number: 0
  script:  -m pip install . --no-deps --no-build-isolation -vv
  skip:  true

requirements:
  host:
    - python
    - pip
    - setuptools >=77.0.3
    - setuptools_scm >=8
  run:
    - python
    - numpy >=1.26
    - packaging >=24.2
    - pandas >=2.2
  run_constrained:

    - scipy >=1.13
    - numbagg >=0.8
    - numba >=0.62
    - flox >=0.9

    - netcdf4 >=1.6.0
    - h5netcdf >=1.4.0
    - zarr >=2.18

    - sparse >=0.15

    - cartopy >=0.23

- scipy >=1.13
- numbagg >=0.9
- numba >=0.62
- flox >=0.10

- netcdf4 >=1.6.0
- h5netcdf >=1.5.0
- zarr >=3.0

- sparse >=0.15

- cartopy >=0.23
- matplotlib-base >=3.8








test:
  source_files:
    - conftest.py
    - pyproject.toml
  imports:
    - xarray
    - xarray.backends
  commands:
    - pip check
    - python -c "from importlib.metadata import version; assert(version('xarray')=='2026.4.0')"
    - pytest . --pyargs xarray.tests  --deselect=test_backends.py::TestNetCDF4Data::test_compression_encoding --deselect=test_backends_datatree.py::TestNetCDF4DataTree::test_compression_encoding --deselect=test_dataset.py::TestDataset::test_repr -vv -m "not (flaky or mypy or network or slow or slow_hypothesis)" -o addopts="--strict-markers --strict-config"
  requires:
    - pip
    - pytest
    - pytest-xdist
    - pytest-timeout
    - pytest-asyncio
    - pooch
    - scipy
    - netcdf4
    - pytz

about:
  home: https://xarray.dev
  license: Apache-2.0
  license_file: LICENSE
  license_family: Apache
  summary: N-D labeled arrays and datasets in Python.
  description: |
    xarray (formerly xray) is an open source project and Python package
    that makes working with labelled multi-dimensional arrays simple,
    efficient, and fun!

    xarray introduces labels in the form of dimensions, coordinates and
    attributes on top of raw NumPy_-like arrays, which allows for a more
    intuitive, more concise, and less error-prone developer experience.
    The package includes a large and growing library of domain-agnostic functions for advanced analytics and visualization with these data structures.

    xarray was inspired by and borrows heavily from pandas_, the popular data analysis package focused on labelled tabular data.
    It is particularly tailored to working with netCDF_ files, which were the source of xarray's data model, and integrates tightly with dask_ for parallel computing.
  dev_url: https://github.com/pydata/xarray
  doc_url: https://docs.xarray.dev

extra:
  recipe-maintainers:
    - jhamman
    - ocefpaf
    - shoyer

INFO:pbpcli.utils.common:Unsetting env var: CONDA_PKGS_DIRS
INFO:pbpcli.utils.common:Unsetting env var: CONDA_SUBDIR
Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 418, in yamlize
    return yaml.load(data, Loader=StringifyNumbersLoader)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/yaml/__init__.py", line 81, in load
    return loader.get_single_data()
           ~~~~~~~~~~~~~~~~~~~~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/yaml/constructor.py", line 49, in get_single_data
    node = self.get_single_node()
  File "yaml/_yaml.pyx", line 674, in yaml._yaml.CParser.get_single_node
  File "yaml/_yaml.pyx", line 688, in yaml._yaml.CParser._compose_document
  File "yaml/_yaml.pyx", line 732, in yaml._yaml.CParser._compose_node
  File "yaml/_yaml.pyx", line 848, in yaml._yaml.CParser._compose_mapping_node
  File "yaml/_yaml.pyx", line 861, in yaml._yaml.CParser._parse_next_event
yaml.parser.ParserError: while parsing a block mapping
  in "<unicode string>", line 4, column 1
did not find expected key
  in "<unicode string>", line 43, column 1

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/opt/conda/envs/webhook-listener/bin/pbp", line 10, in <module>
    sys.exit(cli())
             ~~~^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1442, in __call__
    return self.main(*args, **kwargs)
           ~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1363, in main
    rv = self.invoke(ctx)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1830, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 1226, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/click/core.py", line 794, in invoke
    return callback(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 430, in wrapper
    return func(*args, **kwargs)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 700, in from_feedstock
    build_configs, test_configs = expand_build_matrix(
                                  ~~~~~~~~~~~~~~~~~~~^
        feedstock.recipe_dir_path, src_channels, parsed_user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/cli/generate.py", line 96, in expand_build_matrix
    _targets, build_configs, test_configs = variants_for_feedstock_and_platform(
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^
        recipe_dir_path, platform, src_channels, user_variants
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/pbpcli/utils/expand_variants.py", line 568, in variants_for_feedstock_and_platform
    metadata_tuples = api.render(
        parsed.recipe,
    ...<2 lines>...
        variants=user_variants,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/api.py", line 60, in render
    metadata_tuples = render_recipe(
        recipe_path,
    ...<4 lines>...
        permit_unsatisfiable_variants=permit_unsatisfiable_variants,
    )
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/render.py", line 1048, in render_recipe
    m = MetaData(str(recipe), config=config)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1250, in __init__
    self.parse_again(permit_undefined_jinja=True, allow_no_other_outputs=True)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 1338, in parse_again
    self.meta = parse(
                ~~~~~^
        self._get_contents(
        ^^^^^^^^^^^^^^^^^^^
    ...<5 lines>...
        path=self.meta_path,
        ^^^^^^^^^^^^^^^^^^^^
    )
    ^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 581, in parse
    res = yamlize(data)
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/metadata.py", line 422, in yamlize
    raise UnableToParse(original=e)
          ~~~~~~~~~~~~~^^^^^^^^^^^^
  File "/opt/conda/envs/webhook-listener/lib/python3.13/site-packages/conda_build/exceptions.py", line 22, in __init__
    super().__init__(*args, **kwargs)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
TypeError: CondaError.__init__() missing 1 required positional argument: 'message'
'
aws lambda request id: 23df3714-25ac-5eb0-bba4-5583d44f5062
[View in Grafana](https://grafana.anacondaconnect.com/explore?schemaVersion=1&panes=%7B%2286g%22:%7B%22datasource%22:%22loki-central%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7B__aws_cloudwatch_log_group%3D%5C%22%2Faws%2Flambda%2Fpbp-graph-submitter-prod%5C%22%7D%20%7C%20json%20%7C%20requestId%20%3D%20%6023df3714-25ac-5eb0-bba4-5583d44f5062%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22loki-central%22%7D,%22editorMode%22:%22builder%22,%22direction%22:%22backward%22%7D%5D,%22range%22:%7B%22from%22:%22now-3h%22,%22to%22:%22now%22%7D,%22panelsState%22:%7B%22logs%22:%7B%22columns%22:%7B%220%22:%22Time%22,%221%22:%22Line%22%7D,%22visualisationType%22:%22table%22,%22labelFieldName%22:%22labels%22%7D%7D%7D%7D&orgId=1)</details>

File: recipe/meta.yaml [hunk 1]
```diff
@@ -22,21 +22,19 @@ requirements:
...
-    # numpy 2.4+ introduced behavior changes which may affect usage corner cases (affects visible in
-    # tests: test_curvefit_helpers, test_concat_periods). Added max cap to avoid the issue.
-    - numpy >=1.26,<2.4
-    - packaging >=24.1
+    - numpy >=1.26
+    - packaging >=24.2
     - pandas >=2.2
   run_constrained:
     # accel
     - scipy >=1.13
```
Comment (ELundby45) [inline recipe/meta.yaml]:
```suggestion
    - scipy >=1.15
```
https://github.com/pydata/xarray/blob/v2026.04.0/pyproject.toml#L30

File: recipe/meta.yaml
Comment (ELundby45) [inline recipe/meta.yaml]:
```suggestion
    - matplotlib-base >=3.10
```
    https://github.com/pydata/xarray/blob/v2026.04.0/pyproject.toml#L50

**Review by ELundby45 — APPROVED** [review]
